test/e2e/storage/regional_pd.go

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/kubelet/apis"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	"strings"
+	"time"
+)
+
+const pvDeletionTimeout = 3 * time.Minute
+
+var _ = utils.SIGDescribe("Regional PD [Feature:RegionalPD]", func() {
+	f := framework.NewDefaultFramework("regional-pd")
+
+	// filled in BeforeEach
+	var c clientset.Interface
+	var ns string
+
+	BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+
+		framework.SkipUnlessProviderIs("gce", "gke")
+		framework.SkipUnlessMultizone(c)
+	})
+
+	Describe("RegionalPD", func() {
+		It("should provision storage [Slow]", func() {
+			testVolumeProvisioning(f, c, ns)
+		})
+
+		It("should failover to a different zone when all nodes in one zone become unreachable [Slow] [Disruptive]", func() {
+			testZonalFailover(f, c, ns)
+		})
+
+	})
+})
+
+func testVolumeProvisioning(f *framework.Framework, c clientset.Interface, ns string) {
+	cloudZones := getRandomCloudZones(c)
+
+	// This test checks that dynamic provisioning can provision a volume
+	// that can be used to persist data among pods.
+	test := storageClassTest{
+		name:           "HDD Regional PD on GCE/GKE",
+		cloudProviders: []string{"gce", "gke"},
+		provisioner:    "kubernetes.io/gce-pd",
+		parameters: map[string]string{
+			"type":          "pd-standard",
+			"replica-zones": strings.Join(cloudZones, ","),
+		},
+		claimSize:    "1.5G",
+		expectedSize: "2G",
+		pvCheck: func(volume *v1.PersistentVolume) error {
+			return checkGCEPD(volume, "pd-standard")
```

Comment:
Do you also want to verify that the PD has the correct zone labels on it?  Also, any way we can validate from cloud provider that it's a RePD?

Reply:
Will add code to check zone labels. No I can't think of an easy way to check for RePDs. `GCECloud` doesn't expose a `GetDisk()` today (maybe it should for testing purposes?).